datumaro/plugins/synthetic_data/image_generator.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (C) 2022 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import logging as log
+import os
+import os.path as osp
+from importlib.resources import read_text
+from multiprocessing import Pool
+from random import Random
+from typing import List, Optional, Tuple
+
+import cv2 as cv
+import numpy as np
+
+from datumaro.components.dataset_generator import DatasetGenerator
+
+from .utils import IFSFunction, augment, colorize, download_colorization_model
+
+
+class ImageGenerator(DatasetGenerator):
+    """
+    ImageGenerator generates 3-channel synthetic images with provided shape.
+    """
+
+    def __init__(self, output_dir: str, count: int, shape: Tuple[int, int]):
+        super().__init__(output_dir, count, shape)
+        assert len(self._shape) == 2
+        self._height, self._width = self._shape
+
+        if self._height < 13 or self._width < 13:
+            raise ValueError(
+                "Image generation with height or width of less than 13 is not supported"
+            )
+
+        self._weights = self._create_weights(IFSFunction.NUM_PARAMS)
+        self._threshold = 0.2
+        self._iterations = 200000
+
+        self._path = os.getcwd()
+        download_colorization_model(self._path)
```

Comment:
Probably, it would be useful to add a message about model downloading in the log.

Reply:
done